backend/tracim_backend/lib/core/storage.py

```diff
@@ -0,0 +1,308 @@
+from contextlib import contextmanager
+from datetime import datetime
+import os
+import tempfile
+import typing
+
+from depot.fields.upload import UploadedFile
+from depot.io.interfaces import StoredFile
+from depot.manager import DepotManager
+import filelock
+from hapic.data import HapicFile
+from preview_generator.exception import UnavailablePreviewType
+from preview_generator.exception import UnsupportedMimeType
+from preview_generator.manager import PreviewManager
+
+from tracim_backend import CFG
+from tracim_backend.config import DepotFileStorageType
+from tracim_backend.exceptions import CannotGetDepotFileDepotCorrupted
+from tracim_backend.exceptions import PageOfPreviewNotFound
+from tracim_backend.exceptions import PreviewGeneratorPassthroughError
+from tracim_backend.exceptions import TracimUnavailablePreviewType
+from tracim_backend.exceptions import UnavailablePreview
+from tracim_backend.lib.utils.logger import logger
+
+TEMPORARY_PREFIX = "tracim-revision-content"
+
+
+class StorageLib:
+    """
+    Helper class to handle more easily usage with depot stored file
+    """
+
+    def __init__(self, app_config: CFG) -> None:
+        self.app_config = app_config
+        self.uploaded_file_depot = DepotManager.get(
+            app_config.UPLOADED_FILES__STORAGE__STORAGE_NAME
+        )
+        self.preview_manager = PreviewManager(app_config.PREVIEW_CACHE_DIR, create_folder=True)
+
+    def _get_depot_file(self, depot_file) -> StoredFile:
+        try:
+            return self.uploaded_file_depot.get(depot_file)
+        except IOError as exc:
+            logger.warning(
+                self, "Unable to get content filepath, depot is corrupted", exc_info=True
+            )
+            raise CannotGetDepotFileDepotCorrupted(
+                "depot file {} is not accessible, depot seems corrupted".format(depot_file.file_id)
+            ) from exc
+
+    def get_raw_file(
+        self,
+        depot_file: UploadedFile,
+        filename: str,
+        default_filename: str,
+        force_download: bool = None,
+        last_modified: datetime = None,
+    ) -> HapicFile:
+        file = self._get_depot_file(depot_file)
+        # INFO - G.M - 2019-08-08 - use given filename in all case but none or
+        # "raw", where filename returned will be a custom one.
+        if not filename or filename == "raw":
+            filename = default_filename
+        return HapicFile(
+            file_object=file,
+            mimetype=file.content_type,
+            filename=filename,
+            as_attachment=force_download,
+            content_length=file.content_length,
+            last_modified=last_modified or None,
+        )
```

Comment:
`HapicFile` is pretty HTTP-oriented, shouldn't it be kept in controllers?

Reply:
I thought about it but this mean lot of redundant code to do so. The way api work is very similar.

Reply:
It's needed to simplify thing because HapicFile is not set the same way for preview and depot file, if we do not do this, controller has to be aware that for depot it need to do this way, for preview, it need to do this other way. I don't think that something the controller should have to deal with.
Note also that HapicFile is just a container, to have a file-like object returnable by hapic.